libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/building/BuildingProcessor.kt

```diff
@@ -43,16 +44,30 @@ internal object BuildingProcessor {
         action: BuildingAction.QueryBuildingOnWaypoint
     ): BuildingResult.GetDestination {
         val routeOptions = action.progress.route.routeOptions()
+        val coordinatesList = routeOptions?.coordinatesList()
+        val waypointTargets = routeOptions?.waypointTargetsList()
         val coordinateIndex = action.progress.currentLegProgress?.legIndex!! + 1
-        return routeOptions?.coordinatesList()?.getOrNull(coordinateIndex)?.let { point ->
-            BuildingResult.GetDestination(point)
-        } ?: BuildingResult.GetDestination(null)
+        return when {
+            !waypointTargets.isNullOrEmpty() -> {
+                BuildingResult.GetDestination(waypointTargets[coordinateIndex])
```

Comment:
The waypoint target parameter can be provided only for a subset of route coordinates and skipped for the others. If the `waypointTargets` list is not `null` or empty, it doesn't yet mean that the target elemnt is there for the index of the specified coordinate.

The fallback needs to be implemented for each index separately - first check if there's a target for the specific coordinate index, if not, take the coordinate itself.

Reply:
> The waypoint target parameter can be provided only for a subset of route coordinates and skipped for the others.

Are you saying if the coordinates list as follows:
`RouteOptions.coordinatesList(origin, wp1, wp2, destination)`
then
`waypointTargetList` can be of different size than coordinate list. In this case 
`RouteOptions.coordinatesList(origin, wp1, wp2, destination).waypointTargetList(origin, wp2)`

/cc @LukasPaczos

Reply:
No, it means that the `waypointTargetList` can be: `waypointTargetList(origin, null, wp2, null)`. Basically the `waypointTargetList` can have nullable elements.

Reply:
Okay did some more playing with `waypointTargetList` and `coordinatesList`. Didn't know `waypointTargetList` can be of different size than `coordinatesList`. What is the rationale behind this behavior?

Reply:
Hmm, that should be a bug if you take a look at https://docs.mapbox.com/api/navigation/directions/:
> The number of waypoint_targets must be the same as the number of waypoints, but you can skip a waypoint coordinate and show its position in the list with the ; separator

The list should be of the same size but elements can be nullable.

Reply:
Yeah we have the same doc in `RouteOptions.waypointTargetList`, but let me be thorough before bringing this up. Don't want to create unnecessary noise, as it seems like `approachesList` is also working if not the same size.

Reply:
Ok the above is working on the emulator but not on the real device. So yeah the all the list should be of the same size 👍🏻

Reply:
Done. @LukasPaczos